R/clean.R

```diff
@@ -0,0 +1,82 @@
+#' Clean Rust binaries and package cache.
+#'
+#' Removes Rust binaries (such as `.dll`/`.so` libraries), C wrapper object files,
+#' invokes `cargo clean` to reset cargo target directory
+#' (found by default at `pkg_root/src/rust/target/`).
+#' Useful when Rust code should be recompiled from scratch.
+#' @param path \[ string \] Path to the package root.
+#' @export
+clean <- function(path = ".") {
+  root <- rprojroot::find_package_root_file(path = path)
+
+  rust_folder <- normalizePath(
+    file.path(root, "src", "rust"),
+    winslash = "/",
+    mustWork = FALSE
+  )
+
+  toml_path <- normalizePath(
+    file.path(rust_folder, "Cargo.toml"),
+    winslash = "/",
+    mustWork = FALSE
+  )
+
+  # Note: This should be adjusted if `TARGET_DIR` changes in `Makevars`
+  target_dir <- normalizePath(
+    file.path(rust_folder, "target"),
+    winslash = "/",
+    mustWork = FALSE
+  )
+
+  if (!file.exists(toml_path)) {
+    ui_throw(
+      "Unable to clean binaries.",
+      c(
+        bullet_w(
+          "{.file Cargo.toml} not found in {.path {rust_folder}}."
+        )
+      )
+    )
+  }
+
+  env <- rlang::current_env()
+  cargo_envvars <- get_cargo_envvars()
+
+  args <- c(
+    "clean",
+    glue("--manifest-path={toml_path}"),
+    glue("--target-dir={target_dir}"),
+    if (tty_has_colors()) {
+      "--color=always"
+    } else {
+      "--color=never"
+    },
+    "--quiet"
+  )
+  exec_result <- processx::run(
+    command = "cargo",
+    args = args,
+    echo_cmd = FALSE,
```

Comment:
Hijacking this PR to ask this, but why don't we do a `echo_cmd = !isTRUE(quiet)` here instead?

Reply:
Logical `quiet` is not enough to resolve `--quiet` vs nothing vs `--verbose`, so perhaps we should switch to `verbosity = c("quiet", "default", "verbose")` or something like that?

Reply:
Let's make another issue/pr about this.. Maybe I can provide a draft and we can iterate on it. And maybe aim for it to be merged in before 0.3

Reply:
I agree. There are two places where `run()` is used, so let's discuss it separately.